### PR TITLE
cmd/snap-preseed: check that target path exists and is a directory on --reset

### PR DESCRIPTION
**Pull request by stolowski**
Followup to #9015

The reset logic uses globs to find and remove any preseeding artifacts, but doesn't check that target directory exists. This doesn't cause any errors but may be confusing if wrong path is given by mistake - globs don't match anything and `snap-preseed --reset /invalid/path` returns silently. This PR fixes this.

### CODE REVIEW

File: cmd/snap-preseed/main_test.go [hunk 1]
```diff
@@ -491,4 +491,12 @@ func (s *startPreseedSuite) TestReset(c *C) {
 	// running reset again is ok
 	parser = testParser(c)
 	c.Assert(main.Run(parser, []string{"--reset", tmpDir}), IsNil)
+
+	// reset complains if target directory doesn't exist
+	c.Assert(main.Run(parser, []string{"--reset", "/non/existing/chrootpath"}), ErrorMatches, `cannot reset non-existing directory "/non/existing/chrootpath"`)
```
Comment (bboozzoo) [inline cmd/snap-preseed/main_test.go:496]: With #9015 reset should probably try a relative path too?

Comment (stolowski) [reply cmd/snap-preseed/main_test.go:496]: After #9015 --reset will receive an absolute path, but you're right, it is worth adding an extra test case. I'll update this PR once 9015 lands.

Comment (anonymouse64) [reply cmd/snap-preseed/main_test.go:496]: I can open this followup since Pawel is off now, but I will wait til PR count goes down a bit

Comment (anonymouse64) [reply cmd/snap-preseed/main_test.go:496]: hmm actually since this PR's spread jobs seem to be stuck, I will just push that to this PR

Comment (anonymouse64) [reply cmd/snap-preseed/main_test.go]: PR updated

**Comment by anonymouse64** [comment]
@bboozzoo can you re-review this? I added the requested tests, it's a little more invasive, but it looks simpler if you turn off whitespace diff

**Pull request merged**